libs/features/mas/src/hydrate.js

```diff
@@ -175,16 +285,40 @@ function createSpectrumSwcButton(cta, aemFragmentMapping, isOutline, variant) {
             tabIndex: 0,
             size: aemFragmentMapping.ctas.size ?? 'm',
         },
-        cta,
+        cta.innerHTML,
     );
 
-    spectrumCta.addEventListener('click', (e) => {
-        if (e.target !== cta) {
-            /* c8 ignore next 3 */
+    (async () => {
+        try {
+          await customElements.whenDefined('checkout-button');
+          const CheckoutButtonEl = customElements.get('checkout-button');
+          const checkoutBtn = CheckoutButtonEl.createCheckoutButton({}, cta.innerHTML);
```

Comment:
I think this block causes runtime errors in ah gallery:

mas.js:2370 Failed to initialize checkout-button logic: TypeError: Cannot read properties of null (reading 'setAttribute')
    at mas.js:2370:30403